src/services/cache-config/CacheConfigService.ts

```diff
@@ -303,34 +303,38 @@ export class CacheConfigService {
     let address: string[] = this.getConfigAddress(container.name, container.profile).concat(type + "-configuration", name);
     let builder: CompositeOpBuilder = new CompositeOpBuilder();
     let deferred: ng.IDeferred<void> = this.$q.defer<void>();
+
+    this.verifyConfiguration(config, deferred);
+
     config.template = template;
 
     let createNodeRequest: IDmrRequest = this.createAddOperation(address, config, ["name", "type", "template-name", "is-new-node"]);
 
-    this.dmrService.executePost(createNodeRequest)
-      .then(() => {
-        this.createHelper(builder, address.concat("locking", "LOCKING"), config.locking);
-        this.createHelper(builder, address.concat("memory", this.getMemoryType(config.memory)), config.memory);
-        this.createHelper(builder, address.concat("expiration", "EXPIRATION"), config.expiration);
-        this.createHelper(builder, address.concat("indexing", "INDEXING"), config.indexing);
-        this.createHelper(builder, address.concat("compatibility", "COMPATIBILITY"), config.compatibility);
-        this.createHelper(builder, address.concat("partition-handling", "PARTITION_HANDLING"), config["partition-handling"]);
-        this.createHelper(builder, address.concat("transaction", "TRANSACTION"), config.transaction);
-        this.createHelper(builder, address.concat("state-transfer", "STATE_TRANSFER"), config["state-transfer"]);
-        this.createBackupNodes(builder, address.concat("backup"), config.backup);
-
-        if (this.isSecurityAuthorizationEnabled(config)) {
-          this.updateSecurityAuthorization(config);
-          this.createHelper(builder, address.concat("security", "SECURITY"), config.security);
-          this.createHelper(builder, address.concat("security", "SECURITY", "authorization", "AUTHORIZATION"), config.security.SECURITY.authorization);
-        }
+    if (!isRejected(deferred)) {
```

Comment:
@ryanemerson ok, makes sense. Have a look now. I used try catch because I want any of the verifiers to be able to throw an error. It gets captured and it rejects promise. Otherwise, we have a valid configuration and we proceed with DMR call.